### PR TITLE
Move debug traces for CheckPermission into the response

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: internal/services/v1/permissions.go [hunk 1]
```diff
@@ -94,21 +98,18 @@ func (ps *permissionServer) CheckPermission(ctx context.Context, req *v1.CheckPe
...
 
-		marshaled, merr := protojson.Marshal(converted)
-		if merr != nil {
-			return nil, ps.rewriteError(ctx, merr)
-		}
-
+		// TODO(jschorr): Remove this once the trailer is not longer used by anyone.
 		serr := responsemeta.SetResponseTrailerMetadata(ctx, map[responsemeta.ResponseMetadataTrailerKey]string{
-			responsemeta.DebugInformation: string(marshaled),
+			responsemeta.DebugInformation: `{"note": "debug information has been moved into the response object."}`,
```
Comment (ecordell) [inline internal/services/v1/permissions.go:112]: Maybe this should link to the docs page on clients or something and a note that you need to upgrade the client to see those messages?

Comment (josephschorr) [reply internal/services/v1/permissions.go:112]: I could add a spicedb.dev link?

Comment (ecordell) [reply internal/services/v1/permissions.go:112]: SGTM

Comment (josephschorr) [reply internal/services/v1/permissions.go]: Updated

**Review by ecordell — APPROVED** [review]

**Pull request merged**